### PR TITLE
decrypt gpg encrypted values

### PR DESCRIPTION
**Pull request by ivarmedi**

### CODE REVIEW

File: stack.py [hunk 1]
```diff
@@ -139,3 +142,29 @@ def _parse_stack_cfg(content):
...
+        log.error('could not decrypt string. Using its encrypted representation.')
+        return encrypted_string
+
+
+def _check_encryption(obj):
+    if isinstance(obj, dict):
+        return {key: _check_encryption(val) for key, val in obj.items()}
+    elif isinstance(obj, list):
+        return [_check_encryption(val) for val in obj]
+    elif type(obj) is str and obj.find('-----BEGIN PGP MESSAGE-----') == 0:
```
Comment (ivarmedi) [inline stack.py:168]: I guess this check could happen in `_merge_dict` / `_merge_list` instead.

**Comment by bbinet** [comment]
@ivarmedi rather than updating pillarstack with this specific gpg feature, you should rather create a new ext_pillar that would only be responsible for decrypting gpg pillar data so that it could work for both normal pillar and any other existing ext_pillar.

For more ext_pillar information, please see: https://docs.saltstack.com/en/latest/topics/development/external_pillars.html#ext-pillar

**Comment by bbinet** [comment]
For the records, it seems @ivarmedi has done it in https://github.com/ivarmedi/pillarcrypt

**Comment by ivarmedi** [comment]
@bbinet yes! Thank you for the suggestion, it is indeed a much nicer solution. I had limited understanding on how the pillar data was merged among the various ext_pillars.

**Pull request closed without merging**